### ISSUE TITLE
Typo and general comments on the JS chapter

### ISSUE THREAD

**Issue by xari** [ThinkR-open/building-shiny-apps-workflow#97]
Hey @ColinFay ,

I found a Typo in your text; "`IputValue`", on line 402:

https://github.com/ThinkR-open/building-shiny-apps-workflow/blob/d175e631f3d06ff50bc8014f57d9fa7da38b986f/javascript.Rmd#L402

Also; on line 133, you're missing a closing `"`:

> `$("a[href = 'https://thinkr.fr'])`

https://github.com/ThinkR-open/building-shiny-apps-workflow/blob/d175e631f3d06ff50bc8014f57d9fa7da38b986f/javascript.Rmd#L133

### General feedback

This chapter was helpful for me to understand JS in Shiny better. As a JS developer myself, I'm glad that Shiny has JS interoperability, and I like the functions that Golem provides to make this even more of a breeze. I've yet to dive-deep into JS-in-Shiny, but I liked the way you described the use-case for learning about it:

> Knowing just enough JavaScript can improve the quality of your app: especially when you’ve been using R to render some complex UIs: think conditional panels, simulating a button click from the server, hide and show elements… All these things are good examples of where you should be using JavaScript instead of building more or less complex renderUI or insertUI patterns in your server.

Personally, I would like to understand the performance issues of heavy-reliance on `renderUI()` a little better, but am not too sure where to start. For example, I would like to understand the performance loss of this:

```r
# ui.R
h3(textOutput("caption", inline = TRUE))

# server.R
output$caption <- renderText("Lorem ipsum dolor")
```

vs this:

```r
# ui.R
uiOutput("caption")

# server.R
output$caption <- renderUI(h3("Sit amet consecutor"))
```

### Thoughts on jQuery

I'm still unconvinced about whether pushing jQuery will be helpful to most people. Yes; it ships free with Shiny, but to me the utility of jQuery diminished when ES6 introduced methods like `querySelector()`, and the `fetch()` API.

Yes; jQuery made AJAX and DOM node selection much easier pre-ES6, but I actually think that the more modern JS methods use language that is quite intuitive, and an API that is at least just as easy to work with.

For example:

#### selecting DOM elements

* `document.querySelector("#plop")` vs. `$("#plop")`
* `document.querySelector(".pouet")` vs. `$(".pouet")`
* `document.querySelector("a[href = 'https://thinkr.fr']")` vs. `$("a[href = 'https://thinkr.fr']")`

`querySelector()` also corresponds to it's "sister"-method for finding multiple elements with the same selector, `querySelectorAll()`, while jQuery has the `find()` function for performing the same action.

jQuery does handle selecting by innerHTML simple though (`contains()`).

#### making asynchronous requests

`$.ajax()` and `fetch()` both return a `Promise` and use similar syntax for getting the job done.

---

These examples don't make the case that VanillaJS is necessarily more intuitive than jQuery, but I think that its an interesting discussion to have.

Anyway; I'm looking forward to reading more of your book!

All the best.

**Comment by ColinFay** [comment]
Hey @xari. 

Thank you __very much__ for this very complete and interesting feedback. 
That will help a lot. 

I'll get back to this comment in a few days with answers to these and make improvement to this chapter :)

**Comment by ColinFay** [comment]
Feedbacks integrated in the last book revision. 

+ About performance: question tackled in the JavaScript and in the performance chapter about application caveats. 

+ I've refactored a little bit the JavaScript chapter to include more Vanilla JavaScript. I still think jQuery is valuable but I do agree that ES6  make things easier, so I added more focus on native `document` methods, specifying that users can choose the one they want + information about the browser compatibility, which I think is something most Shiny developers are not really conscious of. 
I think it still makes sense to have jQuery for things like `$('#id').show();` for example, instead of getting into the details of `elem.style.display = 'none';. Same goes for `$().remove()`

That's been a tough chapter to write though, so many things to tell :) 

Thanks again @xari for this feedback!!
